tests/ipam_and_scope_and_pool/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "random_string" "this" {
+  length  = 8
+  upper   = false
+  special = false
+  numeric = false
+}
+
+module "create_ipam" {
+  source = "../.."
+
+  vpc_ipam = {
+    ipam = {
+      name = "default_ipam"
+      operating_regions = [
+        { region_name = "us-east-1" },
+        { region_name = "us-west-1" }
+      ]
+      cascade     = true
+      description = random_string.this.result
+    }
+    scopes = [
+      {
+        name = "high_level_container"
+        #module.create_ipam.aws_vpc_ipam.this[0]
+        ipam_id     = aws_vpc_ipam.this["default_ipam"].id
+        description = random_string.this.result
+        tags = {
+          name = "broker_managed"
+        }
+      },
+    ]
+    pools = [
+      {
+        name                              = "pool_of_cidrs"
+        address_family                    = "ipv4"
+        allocation_default_netmask_length = "16"
+        allocation_min_netmask_length     = "16"
+        allocation_max_netmask_length     = "16"
+        description                       = random_string.this.result
+        ipam_scope_name                   = "private_default_scope"
+        locale                            = "us-east-1"
+        tags = {
+          name = "broker_managed"
+        }
+        ipam_scope_id = module.create_ipam.scope["high_level_container"].id
```

Comment:
The pool config should not pass _both_ `ipam_scope_name` AND `ipam_scope_id`... One or the other... Would be a good precondition to error if both are set.